crates/interpreter/src/exec.rs

```diff
@@ -1409,12 +1425,38 @@ struct Frame<'m> {
     ret: &'m [u8],
     locals: Vec<Val>,
     labels: Vec<Label<'m>>,
+    side_table: &'m [SideTableEntry],
 }
 
 impl<'m> Frame<'m> {
-    fn new(inst_id: usize, arity: usize, ret: &'m [u8], locals: Vec<Val>) -> Self {
+    fn new(
+        inst_id: usize, arity: usize, ret: &'m [u8], locals: Vec<Val>,
+        side_table: &'m [SideTableEntry],
+    ) -> Self {
         let label = Label { arity, kind: LabelKind::Block, values_cnt: 0 };
-        Frame { inst_id, arity, ret, locals, labels: vec![label] }
+        Frame { inst_id, arity, ret, locals, labels: vec![label], side_table }
+    }
+
+    fn skip_jump(&mut self) {
+        self.side_table = Self::jump(self.side_table, 1);
+    }
+
+    fn take_jump(&mut self, parser_pos: &'m [u8], offset: usize) -> Parser<'m> {
+        self.side_table = Self::jump(self.side_table, offset as isize);
+        let entry = self.side_table[0].view();
+        self.side_table = Self::jump(self.side_table, entry.delta_stp as isize);
+        unsafe { Parser::new(Self::jump(parser_pos, entry.delta_ip as isize)) }
+    }
+
+    // TODO(dev/fast-interp): Add debug asserts when `off` is positive and negative, and `toctou`
+    // support.
+    fn jump<T>(cur: &'m [T], off: isize) -> &'m [T] {
```

Comment:
We should rename this function now that it's also used by `skip_jump` for the side-table slice. We should also move it out of `Frame` as a free function. Maybe something like:

```rust
fn offset_front<T>(slice: &[T], count: isize) -> &[T] {
   // ...
}
```